### ISSUE TITLE
debugger in cling

### ISSUE THREAD

**Issue by gouarin** [root-project/cling#292]
We would like to add a debugger in https://github.com/QuantStack/xeus-cling. So we want to attach lldb to the source provided to the JIT. I can see in the source code of cling, you added this possibility

https://github.com/root-project/cling/blob/ee374738072fab73d2b9896c968a000cf0a1bfb4/lib/Interpreter/IncrementalJIT.cpp#L305 

but it's commented.

Could you tell me why ? And if you can give me some feedbacks to be able to have a debugger working on the given code to the cling interpreter ?

cc @SylvainCorlay @JohanMabille

**Comment by Axel-Naumann** [comment]
"Why" is simple: because it needs someone to resurrect this :-) I'd *love* to have it back - would you mind giving it a try? You should be able to look at examples in llvm / clang (maybe `lli` etc) to see how to do it!

**Comment by SylvainCorlay** [comment]
@a

**Comment by SylvainCorlay** [comment]
Thanks. For context we are working on the UI and protocol for debugging with jupyter ans it will soon be usable with the xeus-python kernel.

Since a lot of the stack is common with xeus-cling, hopefully a lot of it will be reused for c++.

By the way, did you guys consider tagging another release? We are waiting for a release including our PR on the metaprocessor to improve our kernel with better support for console workflows.

**Comment by Axel-Naumann** [comment]
Yes, [v0.6 is finally released](https://github.com/root-project/cling/releases/tag/v0.6) - thanks to @vgvassilev !
